### ISSUE TITLE
bug: `coder_paramter` is empty when `type = "number"` and used with `option`

### ISSUE THREAD

**Issue by matifali** [coder/terraform-provider-coder#102]
Version: 0.6.13

# Problem
`option` does not work when `type = "number"` when we use `option`, we can not set the `default` and hence the `coder_parameter` of `type = "string"` has an empty value and terraform plan does not work.

# Example
```hcl
data "coder_parameter" "cpu" {
  name        = "CPU"
  description = "Choose CPU for your workspace? (min: 4 cores, max: 16 cores)"
  type        = "number"
  mutable     = false
  option {
    name  = "4 cores"
    value = "04"
  }
  option {
    name  = "8 cores"
    value = "08"
  }
  option {
    name  = "16 cores"
    value = "16"
  }
}

resource "docker_container" "workspace" {
  count      = data.coder_workspace.me.start_count
  image      = "ubuntu:latest"
  cpu_shares = data.coder_parameter.cpu.value
  memory     = 32768
  # Uses lower() to avoid Docker restriction on container names.
  name = "coder-${data.coder_workspace.me.owner}-${lower(data.coder_workspace.me.name)}"
  # Hostname makes the shell more user friendly: coder@my-workspace:~$
  hostname = lower(data.coder_workspace.me.name)
  dns      = ["1.1.1.1"]
  # Use the docker gateway if the access URL is 127.0.0.1
  command = ["sh", "-c", replace(coder_agent.main.init_script, "127.0.0.1", "host.docker.internal")]
  env     = ["CODER_AGENT_TOKEN=${coder_agent.main.token}"]
}
```
# Solution

Allow setting `default` along with 'option' to force a default option.
Or
Create a new nested property `default` of type `bool` under `option` to make that `option` the default choice.

**Comment by matifali** [comment]
related to https://github.com/coder/coder/issues/6328

**Comment by matifali** [comment]
resolved after #104 is merged and `0.6.14` is released.
